daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala

```diff
@@ -441,16 +441,38 @@ class IdeClient(val compiledPackages: CompiledPackages) extends ScriptLedgerClie
                   result = Success(Right(results.toSeq))
               }
           }
+        case SResultFinalValue(v) =>
+          // The final result should always be unit.
+          result = Failure(new RuntimeException(s"Unexpected non-unit final result: $v"))
         case SResultScenarioCommit(_, _, _, _) =>
           result = Failure(
             new RuntimeException("FATAL: Encountered scenario commit in DAML Script"))
         case SResultError(err) =>
           // Capture the error and exit.
           result = Failure(err)
-        case err =>
-          // TODO: Figure out when we hit this
-          // Capture the error (but not as SError) and exit.
-          result = Failure(new RuntimeException(s"FAILED: $err"))
+        case SResultNeedTime(callback @ _) =>
+          result = Failure(
+            new RuntimeException(
+              "FATAL: GetTime should have been handled by the DAML Script runner"))
+        case SResultNeedPackage(pkg, callback @ _) =>
+          result = Failure(
+            new RuntimeException(
+              s"FATAL: Missing package $pkg should have been reported at Script compilation"))
+        case SResultScenarioInsertMustFail(committers @ _, optLocation @ _) =>
```

Comment:
It you want to be consistent the errors from here on could all start with `Encountered scenario instruction`.

Reply:
Good point, I've made the messages more consistent.